File: src/test/java/edu/kit/iti/formal/stvs/model/table/SpecificationColumnTest.java
package edu.kit.iti.formal.stvs.model.table;

import org.junit.Before;
import org.junit.Test;

import java.util.Arrays;

import static org.junit.Assert.*;

/**
 * Created by bal on 26.02.17.
 */
public class SpecificationColumnTest {
  private SpecificationColumn<Integer> column;

  @Before
  public void setUp() {
    column = new SpecificationColumn<>(Arrays.asList(1,2,3,4,5));
  }

  @Test
  public void testEquals() throws Exception {
    SpecificationColumn identical = new SpecificationColumn<>(Arrays.asList(1,2,3,4,5));
    assertEquals(identical, column);
    column.setComment("Comment");
    assertNotEquals(identical, column);
    identical.setComment("Comment");
    assertEquals(identical, column);
    identical.setComment("Different comment");
    assertNotEquals(identical, column);
    assertNotEquals(null, column);
    assertEquals(column, column);
  }

  @Test
  public void testGetSetComment() {
    assertEquals("", column.getComment());
    column.setComment("Comment");
    assertEquals("Comment", column.getComment());
    column.setComment("NewComment");
    assertEquals("NewComment", column.getComment());
  }

  @Test
  public void testToString() throws Exception {
    column.setComment("Comment");
    assertEquals("SpecificationColumn(cells: [1, 2, 3, 4, 5], comment: Comment)",
        column.toString());
  }
}

File: stvs/src/test/kotlin/edu/kit/iti/formal/stvs/model/table/SpecificationColumnTest.kt
/* *****************************************************************
 * This file belongs to verifaps-lib (https://verifaps.github.io).
 * SPDX-License-Header: GPL-3.0-or-later
 * 
 * This program isType free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program isType distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a clone of the GNU General Public
 * License along with this program.  If not, see
 * <http://www.gnu.org/licenses/gpl-3.0.html>.
 * *****************************************************************/
package edu.kit.iti.formal.stvs.model.table

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

/**
 * Created by bal on 26.02.17.
 */
class SpecificationColumnTest {
    private val column = SpecificationColumn(mutableListOf(1, 2, 3, 4, 5))

    @Test
    fun testEquals() {
        val column = SpecificationColumn(mutableListOf(1, 2, 3, 4, 5))
        val identical: SpecificationColumn<*> = SpecificationColumn(mutableListOf(1, 2, 3, 4, 5))
        Assertions.assertEquals(identical, column)
        column.comment = "Comment"
        Assertions.assertNotEquals(identical, column)
        identical.comment = "Comment"
        Assertions.assertEquals(identical, column)
        identical.comment = "Different comment"
        Assertions.assertNotEquals(identical, column)
        Assertions.assertNotEquals(null, column)
        Assertions.assertEquals(column, column)
    }

    @Test
    fun testGetSetComment() {
        Assertions.assertEquals("", column.comment)
        column.comment = "Comment"
        Assertions.assertEquals("Comment", column.comment)
        column.comment = "NewComment"
        Assertions.assertEquals("NewComment", column.comment)
    }

    @Test
    fun testToString() {
        column.comment = "Comment"
        Assertions.assertEquals("SpecificationColumn(cells: [1, 2, 3, 4, 5], comment: Comment)", column.toString())
    }
}